src/asynch.rs

```diff
@@ -20,9 +20,9 @@ where
     T: Send + 'static,
 {
     let result = std::sync::Arc::new(std::sync::Mutex::new(None));
+    let stack_size = ExecutionState::with(|s| s.config.stack_size);
     let task_id = ExecutionState::spawn(
-        Wrapper::new(fut, std::sync::Arc::clone(&result)),
-        TaskType::Future,
+        AtomicTask::from_future(stack_size, Box::pin(Wrapper::new(fut, std::sync::Arc::clone(&result)))),
```

Comment:
if you make the change I suggest below about removing the BoxFuture from `from_future`, you can remove `Box::pin` here

Reply:
If we do this, we still have to `Box::pin` inside `from_future` right?

Reply:
Right, but that makes a little more sense from an abstraction perspective -- the caller doesn't care whether the future is pinned, only the implementation does.